53-ec2-read-ahead-kb.rules

```diff
@@ -0,0 +1,9 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the MIT License. See the LICENSE accompanying this file
+# for the specific language governing permissions and limitations under
+# the License.
+
+
+# increase read_ahead_kb when a nfs/efs export is mounted to improve throughput 
+SUBSYSTEM=="bdi", ACTION=="add", PROGRAM="/bin/awk -v bdi=$kernel 'BEGIN{ret=1} {if ($4 == bdi) {ret=0}} END{exit ret}' /proc/fs/nfsfs/volumes", ATTR{read_ahead_kb}="15360"
```

Comment:
Current versions of systemd-udevd require the `$4` awk variable to be escaped, otherwise udev reports the following error:

```
/usr/lib/udev/rules.d/53-ec2-read-ahead-kb.rules:9 Invalid value "/bin/awk -v bdi=$kernel 'BEGIN{ret=1} {if (\$4 == bdi) {ret=0}} END{exit ret}' /proc/fs/nfsfs/volumes" for PROGRAM (char 45: invalid substitution type), ignoring, but please fix it.
```

It should be written as `$$4` instead. Note that I haven't tested this on older versions of udev is legacy distributions.

Reply:
I looked a bit deeper and it's systemd 243 (more than 2 years old at this point) that introduces the hard requirement on the `$$` syntax. So I'd recommend using `$$4` here and letting distros relying on older systemd releases (like Amazon Linux 2) patch as necessary.